spec/system/admin/order_spec.rb

```diff
@@ -571,6 +571,51 @@ def new_order_with_distribution(distributor, order_cycle)
 
           expect(page).to have_content "Shipping: #{different_shipping_method_for_distributor1.name}"
         end
+
+        context "when the distributor unsupport a shipping method that's selected in an existing order " do
+          before do
+            distributor1.shipping_methods = [shipping_method_for_distributor1, different_shipping_method_for_distributor1]
+            order.shipments.each(&:refresh_rates)
+            order.select_shipping_method(different_shipping_method_for_distributor1)
+            distributor1.shipping_methods = [shipping_method_for_distributor1]
+          end
+          context "shipment is shipped" do
+            before do
+              order.shipments.first.update_attribute(:state, 'shipped')
```

Comment:
...when the shipment is updated as shipped it is actually that pre-determined shipment which is being considered:
```

=> #<Spree::Shipment:0x00007f5d87308680
 id: 1586,
 tracking: "U10000",
 number: "100",
 cost: 0.0,
 shipped_at: nil,
 order_id: 1920,
 address_id: 10846,
 created_at: Tue, 24 Jan 2023 03:39:18.117486000 AEDT +11:00,
 updated_at: Tue, 24 Jan 2023 03:52:09.860837000 AEDT +11:00,
 state: "shipped",
 stock_location_id: 672,
 included_tax_total: 0.0,
 additional_tax_total: 0.0,
 adjustment_total: 0.0>
```

And this one has indeed  `cost: 0.0`.

I'm not sure what the best way to go around this could be, but perhaps:
- creating a separate shipment and associating it with the intended shipping method, something like:
`let!(:shipment) { create(:shipment_with, :shipping_method, shipping_method: different_shipping_method_for_distributor1) }`

- And then associate this with `order`
- And only after that 'ship' the order

Or:
- create an incomplete order for this purpose (instead of recycling `order`), and then "process" the order like `./spec/system/admin/reports_spec.rb:208`

I did not manage to bring this to work right now, but I hope this comment already helps.